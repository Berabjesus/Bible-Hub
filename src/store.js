import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';

const middlewares = [thunk];
const store = createStore(rootReducer, applyMiddleware(...middlewares));

export default store;
