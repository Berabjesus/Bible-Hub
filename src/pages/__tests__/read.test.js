import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Read from "../read";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../../store";

describe("Bible page", () => {
  it("matches the snapshot", () => {
    const location = {
      state: {
        id: "test",
      },
    };
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Read location={location} />
        </Provider>
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
