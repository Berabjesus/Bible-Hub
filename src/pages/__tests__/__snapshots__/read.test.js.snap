// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bible page matches the snapshot 1`] = `
<section>
  <aside
    className="row d-flex pb-5 aside slideRight"
  >
    <div
      className="flex-column text-white col-9 passage d-flex"
    >
      <small
        className="mx-auto mb-1 pb-1 border-bottom w-100 text-center"
      >
        test
        Books
      </small>
    </div>
    <div
      className="col-3 flex-column text-white text-center border-left border-light px-1 d-flex passage"
    >
      <small
        className="mx-auto mb-1 pb-1 border-bottom w-100 text-center"
      >
        Ch
      </small>
    </div>
    <button
      className=" align-self-center toggle"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 "
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          Object {
            "color": "White",
            "fontSize": "15px",
          }
        }
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </aside>
  <article
    className="pl-3 col-lg-8 mx-auto pr-1 article"
    style={
      Object {
        "background": "white",
        "color": "#100C1A",
      }
    }
  />
  <div
    className="d-flex justify-content-between py-4 pr-2 pl-1 align-items-center bottom"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-moon fa-w-16 "
        data-icon="moon"
        data-prefix="far"
        focusable="false"
        role="img"
        style={
          Object {
            "color": "White",
            "fontSize": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M279.135 512c78.756 0 150.982-35.804 198.844-94.775 28.27-34.831-2.558-85.722-46.249-77.401-82.348 15.683-158.272-47.268-158.272-130.792 0-48.424 26.06-92.292 67.434-115.836 38.745-22.05 28.999-80.788-15.022-88.919A257.936 257.936 0 0 0 279.135 0c-141.36 0-256 114.575-256 256 0 141.36 114.576 256 256 256zm0-464c12.985 0 25.689 1.201 38.016 3.478-54.76 31.163-91.693 90.042-91.693 157.554 0 113.848 103.641 199.2 215.252 177.944C402.574 433.964 344.366 464 279.135 464c-114.875 0-208-93.125-208-208s93.125-208 208-208z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <small
        className="text-white d-block"
      >
        Dark
      </small>
    </button>
    <a
      className="d-flex flex-column align-items-center"
      href="/"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-home fa-w-18 "
        data-icon="home"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          Object {
            "color": "White",
            "fontSize": "18px",
          }
        }
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <small>
        Home
      </small>
    </a>
    <a
      className="d-flex flex-column align-items-center"
      href="/favorites"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-list fa-w-16 "
        data-icon="list"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          Object {
            "color": "White",
            "fontSize": "18px",
          }
        }
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M80 368H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm0-320H16A16 16 0 0 0 0 64v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16V64a16 16 0 0 0-16-16zm0 160H16a16 16 0 0 0-16 16v64a16 16 0 0 0 16 16h64a16 16 0 0 0 16-16v-64a16 16 0 0 0-16-16zm416 176H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <small>
        List
      </small>
    </a>
  </div>
</section>
`;
